src/backend/InvenTree/part/models.py

```diff
@@ -661,6 +661,49 @@ def validate_ipn(self, raise_error=True):
             if match is None:
                 raise ValidationError(_(f'IPN must match regex pattern {pattern}'))
 
+    def validate_revision(self):
+        """Check the 'revision' and 'revision_of' fields."""
+        # Part cannot be a revision of itself
+        if self.revision_of:
+            if self.revision_of == self:
+                raise ValidationError({
+                    'revision_of': _('Part cannot be a revision of itself')
+                })
+
+            # Part cannot be a revision of a part which is itself a revision
+            if self.revision_of.revision_of:
+                raise ValidationError({
+                    'revision_of': _(
+                        'Cannot make a revision of a part which is already a revision'
+                    )
+                })
```

Comment:
I think having trees of revisions would be helpful

Reply:
I thought about this, but having a potentially recursive relationship here makes it very hard to determine the total number of revisions for a part. This is why we utilize `MPTT` for the part variant - but we cannot have a second tree structure attached to the part model.

The limitation here is not too restrictive ( I think ) - as long as you point the "revision_of" to the same base part, you can have as many revisions as you like.

Feel free to suggest another approach!

Reply:
I think from a user perspective it is limiting as you need to manually re-structure multiple parts to achieve the same result. But I can see this taking a lot of effort to include in the PR.